### PR TITLE
SubscribeSafe

### PR DESCRIPTION
**Pull request by helto4real**
## Breaking change

Changes behaviour of SubscribeAsync and SubscribeAsyncConcurrent to safely recover from unhandled exception. This was the intended behaviour from the beginning so technically it is a bugfix.

## Proposed change

This PR adds a safe subscription. Safe means it will recover from unhandled exceptions and continue process messages. Normal Subscribe would unsubscribe all subscribers when unhandled exceptions occurs.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/HassModel/NetDeamon.HassModel/ObservableExtensions.cs [hunk 1]
```diff
@@ -69,7 +106,7 @@ private static T HandleTaskSafe<T>(Action<T> onNext, T e, Action<Exception>? onE
             else
             {
                 using var loggerFactory = LoggerFactory.Create(x => { x.AddConsole(); });
```
Comment (BeeHiveJava) [inline src/HassModel/NetDeamon.HassModel/ObservableExtensions.cs:108]: Would be nice to only create the loggerFactory if it's needed - if you pass a logger, you won't need to create it.

Comment (skotl) [reply src/HassModel/NetDeamon.HassModel/ObservableExtensions.cs:108]: Agree, also the error is only logged when the errorhandler is null, which can create confusion: if I passed an error handler _and_  a logger, I'd expect both to be honoured but in this case only the error handler would be.

I must admit to being unsure on the exact logic but perhaps it's like this?

1. If an error handler is passed, call it
2. If a logger is passed, call it
3. If neither, then create a default logger and log a generic message

1.

Comment (helto4real) [reply src/HassModel/NetDeamon.HassModel/ObservableExtensions.cs:108]: I did add overload method just to pass a logger now.  @BeeHiveJava 
1. You not provide anything _> use default factory
2. You provide logger, use that
3. You provee callback, use that.
They cannot be used in combination.

Comment (helto4real) [reply src/HassModel/NetDeamon.HassModel/ObservableExtensions.cs:108]: Ohh I realize now I forgot to add that logic :) haha good eye!

**Comment by helto4real** [comment]
Merging this due to low activity on reviews

**Pull request merged**